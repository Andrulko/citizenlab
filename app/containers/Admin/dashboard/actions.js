/*
 *
 * AdminPage/DashboardPage actions
 *
 */

import {
  LOAD_USERS_REPORT_REQUEST, LOAD_USERS_REPORT_SUCCESS, LOAD_USERS_REPORT_ERROR, LOAD_IDEA_TOPICS_REPORT_REQUEST,
  LOAD_IDEA_TOPICS_REPORT_SUCCESS, LOAD_IDEA_TOPICS_REPORT_ERROR, LOAD_IDEA_AREAS_REPORT_REQUEST,
  LOAD_IDEA_AREAS_REPORT_SUCCESS, LOAD_IDEA_AREAS_REPORT_ERROR,
} from './constants';

export function loadUsersReportRequest(startAt, endAt, interval) {
  return {
    type: LOAD_USERS_REPORT_REQUEST,
    startAt,
    endAt,
    interval,
  };
}

export function loadUsersReportSuccess(response) {
  return {
    type: LOAD_USERS_REPORT_SUCCESS,
    payload: response,
  };
}

export function loadUsersReportError(error) {
  return {
    type: LOAD_USERS_REPORT_ERROR,
    payload: error,
  };
}

export function loadIdeaTopicsReportRequest(startAt, endAt) {
  return {
    type: LOAD_IDEA_TOPICS_REPORT_REQUEST,
    startAt,
    endAt,
  };
}

export function loadIdeaTopicsReportSuccess(response) {
  if (response && response.data && response.topics) {
    return {
      type: LOAD_IDEA_TOPICS_REPORT_SUCCESS,
      payload: {
        values: response.data,
        labels: response.topics,
      },
    };
  }
  return loadIdeaTopicsReportError('invalid response');
}

export function loadIdeaTopicsReportError(error) {
  return {
    type: LOAD_IDEA_TOPICS_REPORT_ERROR,
    payload: error,
  };
}


export function loadIdeaAreasReportRequest(startAt, endAt) {
  return {
    type: LOAD_IDEA_AREAS_REPORT_REQUEST,
    startAt,
    endAt,
  };
}

export function loadIdeaAreasReportSuccess(response) {
  if (response && response.data && response.areas) {
    return {
      type: LOAD_IDEA_AREAS_REPORT_SUCCESS,
      payload: {
        values: response.data,
        labels: response.areas,
      },
    };
  }
  return loadIdeaAreasReportError('invalid response');
}

export function loadIdeaAreasReportError(error) {
  return {
    type: LOAD_IDEA_AREAS_REPORT_ERROR,
    payload: error,
  };
}
