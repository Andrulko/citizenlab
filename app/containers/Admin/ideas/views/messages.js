/*
 * AdminPage.DashboardPage Messages
 *
 * This contains all the text for the AdminPage.DashboardPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  headerIndex: {
    id: 'app.containers.AdminPage.Ideas.Views.all.headerIndex',
    defaultMessage: 'All Ideas',
  },
  deleteButton: {
    id: 'app.containers.AdminPage.Ideas.Views.all.deleteButton',
    defaultMessage: 'Delete',
  },
  updateButton: {
    id: 'app.containers.AdminPage.Ideas.Views.all.updateButton',
    defaultMessage: 'Edit',
  },
  createButton: {
    id: 'app.containers.AdminPage.Ideas.Views.all.createButton',
    defaultMessage: 'New Idea',
  },
  publishButton: {
    id: 'app.containers.AdminPage.Ideas.Views.form.publishButton',
    defaultMessage: 'Publish Idea',
  },

  ideaLoadingMessage: {
    id: 'app.containers.AdminPage.Ideas.Views.edit.ideaLoadingMessage',
    defaultMessage: 'Loading Idea...',
  },
  ideaLoadingError: {
    id: 'app.containers.AdminPage.Ideas.Views.edit.ideaLoadingError',
    defaultMessage: 'Idea not found!',
  },

});
