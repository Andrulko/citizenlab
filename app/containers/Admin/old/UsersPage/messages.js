/*
 * AdminPage.UsersPage Messages
 *
 * This contains all the text for the AdminPage.UsersPage component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  helmetTitle: {
    id: 'app.containers.AdminPage.UsersPage.helmetTitle',
    defaultMessage: 'Users admin page',
  },
  helmetDescription: {
    id: 'app.containers.AdminPage.UsersPage.helmetDescription',
    defaultMessage: 'List of users in admin backoffice',
  },
});
