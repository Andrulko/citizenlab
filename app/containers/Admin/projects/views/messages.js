/*
 * AdminPage.DashboardPage Messages
 *
 * This contains all the text for the AdminPage.DashboardPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  headerIndex: {
    id: 'app.containers.AdminPage.Projects.Views.all.headerIndex',
    defaultMessage: 'All Projects',
  },
  deleteButton: {
    id: 'app.containers.AdminPage.Projects.Views.all.deleteButton',
    defaultMessage: 'Delete',
  },
  updateButton: {
    id: 'app.containers.AdminPage.Projects.Views.all.updateButton',
    defaultMessage: 'Edit',
  },
  createButton: {
    id: 'app.containers.AdminPage.Projects.Views.all.createButton',
    defaultMessage: 'New Project',
  },
  publishButton: {
    id: 'app.containers.AdminPage.Projects.Views.form.publishButton',
    defaultMessage: 'Publish Project',
  },

  projectLoadingMessage: {
    id: 'app.containers.AdminPage.Projects.Views.edit.projectLoadingMessage',
    defaultMessage: 'Loading Project...',
  },
  projectLoadingError: {
    id: 'app.containers.AdminPage.Projects.Views.edit.projectLoadingError',
    defaultMessage: 'Project not found!',
  },
  tableTitle: {
    id: 'app.containers.AdminPage.AreaDashboard.tableTitle',
    defaultMessage: 'Projects',
  },
});
