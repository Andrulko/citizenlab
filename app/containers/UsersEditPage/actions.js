/*
 *
 * UsersEditPage actions
 *
 */

import {
  LOAD_CURRENT_USER_SUCCESS, LOAD_CURRENT_USER_ERROR, STORE_AVATAR, UPDATE_CURRENT_USER_SUCCESS,
  UPDATE_CURRENT_USER_ERROR, UPDATE_CURRENT_USER, STORE_AVATAR_ERROR, UPDATE_USER_LOCALE, STORE_AVATAR_SUCCESS,
} from './constants';

export function loadCurrentUserSuccess(currentUser) {
  const validResponse = currentUser && currentUser.data;
  const currentUserWithAvatar = currentUser;

  if (!validResponse) {
    return {
      type: 'LOAD_CURRENT_USER_ERROR',
    };
  }

  currentUserWithAvatar.data.attributes.avatar =
    (validResponse && currentUserWithAvatar.data.attributes.avatar
      ? currentUserWithAvatar.data.attributes.avatar.medium
      : '');

  return {
    type: LOAD_CURRENT_USER_SUCCESS,
    payload: currentUser.data.attributes,
    userId: currentUser.data.id,
  };
}

export function loadCurrentUserError() {
  return {
    type: LOAD_CURRENT_USER_ERROR,
  };
}

export function updateCurrentUser(currentUser) {
  return {
    type: UPDATE_CURRENT_USER,
    payload: currentUser,
    userId: currentUser.userId,
  };
}

export function updateLocale(userLocale) {
  return {
    type: UPDATE_USER_LOCALE,
    userLocale,
  };
}

export function updateCurrentUserSuccess(currentUser) {
  const validResponse = !!(currentUser && currentUser.data);
  const currentUserWithAvatar = currentUser;

  if (validResponse && !currentUserWithAvatar.data.attributes.avatar) {
    currentUserWithAvatar.data.attributes.avatar = '';
  }

  return {
    type: (validResponse
        ? UPDATE_CURRENT_USER_SUCCESS
        : UPDATE_CURRENT_USER_ERROR),
    payload: validResponse && currentUser.data.attributes,
    userId: currentUser.data.id,
  };
}

export function updateCurrentUserError() {
  return {
    type: UPDATE_CURRENT_USER_ERROR,
  };
}

export function storeAvatar(avatarBase64, userId) {
  return {
    type: STORE_AVATAR,
    avatarBase64,
    userId,
  };
}

export function storeAvatarSuccess() {
  return {
    type: STORE_AVATAR_SUCCESS,
  };
}

export function storeAvatarError() {
  return {
    type: STORE_AVATAR_ERROR,
  };
}
