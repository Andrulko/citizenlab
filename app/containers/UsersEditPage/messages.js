/*
 * UsersEditPage Messages
 *
 * This contains all the text for the UsersEditPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.containers.UsersEditPage.header',
    defaultMessage: 'Profile',
  },
  firstName: {
    id: 'app.containers.UsersEditPage.firstName',
    defaultMessage: 'First Name',
  },
  lastName: {
    id: 'app.containers.UsersEditPage.lastName',
    defaultMessage: 'Last Name',
  },
  email: {
    id: 'app.containers.UsersEditPage.email',
    defaultMessage: 'E-mail',
  },
  gender: {
    id: 'app.containers.UsersEditPage.gender',
    defaultMessage: 'Gender',
  },
  age: {
    id: 'app.containers.UsersEditPage.age',
    defaultMessage: 'Age',
  },
  loading: {
    id: 'app.containers.UsersEditPage.loading',
    defaultMessage: 'Loading...',
  },
  processing: {
    id: 'app.containers.UsersEditPage.processing',
    defaultMessage: 'Sending...',
  },
  loadError: {
    id: 'app.containers.UsersEditPage.loadError',
    defaultMessage: 'Can\'t load existing profile',
  },
  storeError: {
    id: 'app.containers.UsersEditPage.storeError',
    defaultMessage: 'Can\'t store updated profile',
  },
  stored: {
    id: 'app.containers.UsersEditPage.stored',
    defaultMessage: 'Profile stored',
  },
  avatarLoadError: {
    id: 'app.containers.UsersEditPage.avatarLoadError',
    defaultMessage: 'Existing avatar coudln\'t be loaded',
  },
  avatarStoreError: {
    id: 'app.containers.UsersEditPage.avatarStoreError',
    defaultMessage: 'Can\'t change avatar right now',
  },
});
