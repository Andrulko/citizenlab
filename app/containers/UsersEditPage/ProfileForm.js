import React from 'react';
import PropTypes from 'prop-types';
import { actions, LocalForm } from 'react-redux-form';
import { FormattedMessage } from 'react-intl';
import { Grid, Label, Button } from 'semantic-ui-react';
import LocaleChanger from 'components/LocaleChanger';
import FormInput from 'components/FormInput';

import messages from './messages';
import Avatar from './Avatar';

const LabelInputPair = (props) => (
  <div
    style={{
      lineHeight: (props.hidden ? 'inherit' : '4em'),
      height: (props.hidden ? '0' : 'inherit'),
    }}
  >
    {!props.hidden && <Label htmlFor={props.id}>
      <FormattedMessage {...messages[props.id]} />
    </Label>}
    <FormInput id={props.id} hidden={props.hidden} />
  </div>
);

LabelInputPair.propTypes = {
  id: PropTypes.string.isRequired,
  hidden: PropTypes.bool,
};

export default class ProfileForm extends React.PureComponent {
  componentWillReceiveProps(nextProps) {
    const user = nextProps.userData;

    if (user) {
      // load initial value (unless form being re-rendered)
      this.localFormDispatch(actions.load('user', user));
    }
  }

  render() {
    const { avatarUploadError, avatarUpload, userData, onLocaleChangeClick } = this.props;
    const userLocale = (userData
      ? userData.locale
      : 'en');

    return (
      <LocalForm
        model="user"
        getDispatch={(dispatch) => {
          this.localFormDispatch = dispatch;
        }}
        onSubmit={this.props.onFormSubmit}
      >
        <Grid>
          <Grid.Row columns={2}>
            <Grid.Column width={8}>
              <LabelInputPair id="first_name" />
              <LabelInputPair id="last_name" />
              <LabelInputPair id="email" />
            </Grid.Column>
            <Grid.Column width={8}>
              {userData && userData.userId && <Avatar
                onAvatarUpload={avatarUpload}
                avatarUploadError={avatarUploadError}
                avatarURL={userData.avatar}
                userId={userData.userId}
              />}
              <LocaleChanger
                onLocaleChangeClick={onLocaleChangeClick}
                userLocale={userLocale}
              />
            </Grid.Column>
          </Grid.Row>
        </Grid>
        <Button>
          <FormattedMessage {...messages.submit} />
        </Button>
        <LabelInputPair id="avatar" hidden />
        <LabelInputPair id="locale" hidden />
        <LabelInputPair id="userId" hidden />
      </LocalForm>
    );
  }
}

ProfileForm.propTypes = {
  onFormSubmit: PropTypes.func.isRequired,
  avatarUpload: PropTypes.func.isRequired,
  onLocaleChangeClick: PropTypes.func.isRequired,
  userData: PropTypes.object,
  avatarUploadError: PropTypes.bool,
};
