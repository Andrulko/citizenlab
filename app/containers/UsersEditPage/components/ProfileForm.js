import React from 'react';
import PropTypes from 'prop-types';
import Button from 'components/UI/Button';
import { Grid, Radio } from 'semantic-ui-react';
import { appLocalePairs } from 'i18n';
import ImmutablePropTypes from 'react-immutable-proptypes';
import messages from '../messages';
import Avatar from './Avatar';
import generateErrorsObject from 'components/forms/generateErrorsObject';
import _ from 'lodash';
import { connect } from 'react-redux';
import { makeSelectSetting } from 'utils/tenant/selectors';
import { createStructuredSelector } from 'reselect';
import Input from 'components/UI/Input';
import Select from 'components/UI/Select';
import styled from 'styled-components';
import scrollToComponent from 'react-scroll-to-component';
import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
import LabelWithTooltip from './LabelWithTooltip';
import TextArea from 'components/UI/TextArea';
import { injectTFunc } from 'containers/T/utils';
import moment from 'moment';
import { media } from 'utils/styleUtils';

const NavItemStyled = styled.button`
  display: block;
  height: 26px;
  font-size: 18px;
  font-weight: 500;
  text-align: left;
  color: #222222;
`;

const Nav = ({ goTo }) => (<div>
  <NavItemStyled onClick={() => goTo('h1')}>
    <FormattedMessage {...messages.h1} />
  </NavItemStyled>
  <NavItemStyled onClick={() => goTo('h2')}>
    <FormattedMessage {...messages.h2} />
  </NavItemStyled>
  <NavItemStyled onClick={() => goTo('h3')}>
    <FormattedMessage {...messages.h3} />
  </NavItemStyled>
</div>);

Nav.propTypes = {
  goTo: PropTypes.func.isRequired,
};

class ProfileForm extends React.PureComponent {
  constructor() {
    super();

    this.state = {
      user: null,
      avatar: '',
    };
  }

  componentWillReceiveProps(nextProps) {
    const user = nextProps.userData;

    // if the user wasn't loaded already ...
    if (!this.state.user || _.isEmpty(this.state.user)) {
      this.setState({
        user,
      });
    }

    this.setState({
      avatar: user && user.avatar,
    });
  }

  getToggleValue = (property) => {
    const { user } = this.state;
    if (!user) {
      return false;
    }
    return user[property];
  };

  goToSection = (which) => {
    if (which === 'h1') {
      scrollToComponent(this['section-basics']);
    } else if (which === 'h2') {
      scrollToComponent(this['section-details']);
    } else if (which === 'h3') {
      scrollToComponent(this['section-notifications']);
    }
  };

  handleAvatarUpload = (avatarBase64, userId) => {
    this.props.avatarUpload(avatarBase64, userId);
  };

  // Input, Textarea
  handleInputChange = (value, name) => {
    const { user } = _.clone(this.state);

    // TODO: add validation (tracking errors in component's state will do)
    user[name] = value;
    this.setState({
      user,
    });
  };

  // Select
  handleSelectChange = (option, name) => {
    const { user } = _.clone(this.state);

    // TODO: add validation (tracking errors in component's state will do)
    user[name] = option.value;
    this.setState({
      user,
    });
  };

  // Toggle
  handleToggleChange = (name) => {
    const { user } = _.clone(this.state);

    // TODO: add validation (tracking errors in component's state will do)
    user[name] = !user[name];
    this.setState({
      user,
    });
  };

  handleSubmit(e) {
    e.preventDefault();
    const { user } = this.state;
    const { userData } = this.props;

    const userWithLocale = Object.assign({}, user);
    userWithLocale.locale = userData.locale;

    this.props.onFormSubmit(userWithLocale);
  }

  render() {
    const {
      avatarUploadError, userData: user, onLocaleChangeClick, className,
      processing, storeErrors, locales, intl, tFunc,
    } = this.props;
    const userLocale = (user
      ? user.locale
      : 'en');
    const localesJS = locales.toJS();

    const userErrors = storeErrors && generateErrorsObject(storeErrors);

    /*
     * styled components
    */
    const LeftColumnStyled = styled(Grid.Column)`
      ${media.phone`
        display: none !important;
      `}
    `;

    const RightColumnStyled = styled(Grid.Column)`
      border-radius: 5px;
      background-color: #ffffff;
    `;

    const InputGroupStyled = styled.div`
      margin-top: 40px;
    `;

    const SectionHeaderStyled = styled.div`
      font-size: 25px;
      font-weight: bold;
      text-align: left;
      color: #222222;
    `;

    const SectionSubHeaderStyled = styled.div`
      font-size: 16px;
      text-align: left;
      color: #6b6b6b;
    `;

    const LabelInputPairStyled = styled.div`
      margin-top: 10px;
    `;

    const SectionSeparatorStyled = styled.hr`
      border: none;
      height: 3px;
      width: 100%;
      /* Set the hr color */
      color: #eaeaea; /* old IE */
      background-color: #eaeaea; /* Modern Browsers */
    `;

    const StyledRadio = styled(Radio)`
      label:before {
        /* ! cannot override as important is already set on the styled radio */
        /* TODO: try to fix this */
        background-color: ${(props) => props.checked ? '#3fb57c !important' : 'inherit'};
        border-radius: 500rem;
      }
    `;

    /*
     * props
     */
    const genderOptions = [
      {
        value: 'male',
        label: intl.formatMessage(messages.male),
      },
      {
        value: 'female',
        label: intl.formatMessage(messages.female),
      },
    ];

    const birthYearOptions = [];
    for (let i = parseInt(moment().format('YYYY'), 10); i >= 1900; i -= 1) {
      birthYearOptions.push({
        value: i.toString(),
        label: i.toString(),
      });
    }

    const educationOptions = [];
    for (let i = 0; i <= 8; i += 1) {
      educationOptions.push({
        value: i.toString(),
        label: intl.formatMessage(messages[`ISCED11_${i}`]),
      });
    }

    const areas = [];
    const areaValue = {};
    // TODO: areas from selector (object {value-label} format). same for areaValue

    // TODO: fix data not working properly (when we set input, state is overidden and for some reason passed back as props to ProfileForm, ending in inputs resettings their content. e.g. <empty> -> type a -> set state -> empty again [from props])
    return (<div className={className}>
      <Grid>
        <Grid.Row>
          <LeftColumnStyled width={4}>
            <Nav goTo={this.goToSection} />
          </LeftColumnStyled>
          <RightColumnStyled width={12}>
            {/* BASICS */}
            <section ref={(section1) => { this['section-basics'] = section1; }}>
              <SectionHeaderStyled>
                <FormattedMessage {...messages.h1} />
              </SectionHeaderStyled>
              <SectionSubHeaderStyled>
                <FormattedMessage {...messages.h1sub} />
              </SectionSubHeaderStyled>

              {/* TODO: fix avatar not showing up */}
              {/* TODO: style properly and add icon (+ logic) to trash avatar (i.e. send null) */}
              {user && user.userId && <Avatar
                onAvatarUpload={this.handleAvatarUpload}
                avatarUploadError={avatarUploadError}
                avatarURL={user.avatar}
                userId={user.userId}
              />}

              <InputGroupStyled>
                <LabelInputPairStyled>
                  {/*
                   * TODO: move these recurring blocks to a component with props for re-use
                   */}
                  <LabelWithTooltip id="firstName" hasTooltip />
                  <Input
                    name="firstName"
                    onChange={this.handleInputChange}
                    value={user && user.first_name}
                    error={userErrors && userErrors.first_name && userErrors.first_name[0]}
                  />
                </LabelInputPairStyled>
                <LabelInputPairStyled>
                  <LabelWithTooltip id="lastName" hasTooltip />
                  <Input
                    name="lastName"
                    onChange={this.handleInputChange}
                    value={user && user.last_name}
                    error={userErrors && userErrors.last_name && userErrors.last_name[0]}
                  />
                </LabelInputPairStyled>
                <LabelInputPairStyled>
                  <LabelWithTooltip id="email" hasTooltip />
                  <Input
                    name="email"
                    onChange={this.handleInputChange}
                    value={user && user.email}
                    error={userErrors && userErrors.email && userErrors.email[0]}
                  />
                </LabelInputPairStyled>
                <LabelInputPairStyled>
                  <LabelWithTooltip id="password" hasTooltip />
                  <Input
                    name="password"
                    onChange={this.handleInputChange}
                    value={user && user.password}
                    error={userErrors && userErrors.password && userErrors.password[0]}
                  />
                </LabelInputPairStyled>

                {localesJS && <Select
                  name="locale"
                  onChange={onLocaleChangeClick}
                  value={appLocalePairs.find((l) => l.value === userLocale)}
                  options={appLocalePairs.filter((locale) => localesJS.find((l) => l === locale.value))}
                />}
              </InputGroupStyled>
            </section>

            <SectionSeparatorStyled
              style={{
                margin: '60px 0',
              }}
            />

            {/* DETAILS */}
            <section ref={(section2) => { this['section-details'] = section2; }}>
              <SectionHeaderStyled>
                <FormattedMessage {...messages.h2} />
              </SectionHeaderStyled>
              <SectionSubHeaderStyled>
                <FormattedMessage {...messages.h2sub} />
              </SectionSubHeaderStyled>

              <InputGroupStyled>
                <LabelWithTooltip id="gender" hasTooltip />
                <Select
                  name="gender"
                  placeholder={intl.formatMessage(messages.male)}
                  options={genderOptions}
                  onChange={this.handleSelectChange}
                  value={{
                    value: user.gender,
                    label: user.gender && intl.formatMessage(messages[user.gender]),
                  }}
                  error={userErrors && userErrors.gender[0]}
                />

                <LabelWithTooltip id="bio" hasTooltip />
                <TextArea
                  name="bio_multiloc"
                  onInput={this.handleInputChange}
                  rows={6}
                  placeholder={intl.formatMessage(messages.bio_placeholder)}
                  value={user.bio_multiloc && tFunc(user.bio_multiloc)}
                  error={userErrors && userErrors.bio_multiloc && userErrors.bio_multiloc[0]}
                />

                <LabelWithTooltip id="area" />
                <Select
                  name="area_multiloc"
                  placeholder={intl.formatMessage(messages.area_placeholder)}
                  options={areas}
                  onChange={this.handleSelectChange}
                  value={areaValue}
                  error={userErrors && userErrors.domicilie && userErrors.domicilie[0]}
                />

                <LabelWithTooltip id="birthdate" />
                <Select
                  name="birthdate"
                  options={birthYearOptions}
                  onChange={this.handleSelectChange}
                  value={{
                    value: user.birthyear,
                    label: user.birthyear,
                  }}
                  error={userErrors && userErrors.birthyear && userErrors.birthyear[0]}
                />

                <LabelWithTooltip id="education" />
                <Select
                  name="education"
                  placeholder={intl.formatMessage(messages.area_placeholder)}
                  options={educationOptions}
                  onChange={this.handleSelectChange}
                  value={{
                    value: user.education,
                    label: user.education && intl.formatMessage(messages[user.education]),
                  }}
                  error={userErrors && userErrors.education && userErrors.education[0]}
                />

              </InputGroupStyled>
            </section>

            <SectionSeparatorStyled
              style={{
                margin: '48px 0 25px 0',
              }}
            />

            {/* NOTIFICATIONS */}
            <section ref={(section3) => { this['section-notifications'] = section3; }}>
              <SectionHeaderStyled>
                <FormattedMessage {...messages.h3} />
              </SectionHeaderStyled>
              <SectionSubHeaderStyled>
                <FormattedMessage {...messages.h3sub} />
              </SectionSubHeaderStyled>

              <InputGroupStyled>
                <LabelWithTooltip id="notifications_all_email" isBold />
                <StyledRadio
                  toggle
                  checked={this.getToggleValue('notifications_all_email')}
                  onClick={() => this.handleToggleChange('notifications_all_email')}
                />

                <LabelWithTooltip id="notifications_idea_post" />
                <StyledRadio
                  toggle
                  checked={this.getToggleValue('notifications_idea_post')}
                  onClick={() => this.handleToggleChange('notifications_idea_post')}
                />

                <LabelWithTooltip id="notifications_new_user" />
                <StyledRadio
                  toggle
                  checked={this.getToggleValue('notifications_new_user')}
                  onClick={() => this.handleToggleChange('notifications_new_user')}
                />

                <LabelWithTooltip id="notifications_new_comments" />
                <StyledRadio
                  toggle
                  checked={this.getToggleValue('notifications_new_comments')}
                  onClick={() => this.handleToggleChange('notifications_new_comments')}
                />

                <LabelWithTooltip id="notifications_all_app" isBold />
                <StyledRadio
                  toggle
                  checked={this.getToggleValue('notifications_all_app')}
                  onClick={() => this.handleToggleChange('notifications_all_app')}
                />

                <LabelWithTooltip id="notifications_comment_on_comment" />
                <StyledRadio
                  toggle
                  checked={this.getToggleValue('notifications_comment_on_comment')}
                  onClick={() => this.handleToggleChange('notifications_comment_on_comment')}
                />

                <LabelWithTooltip id="notifications_mention" />
                <StyledRadio
                  toggle
                  checked={this.getToggleValue('notifications_mention')}
                  onClick={() => this.handleToggleChange('notifications_mention')}
                />

                <LabelWithTooltip id="notifications_idea_comment" />
                <StyledRadio
                  toggle
                  checked={this.getToggleValue('notifications_idea_comment')}
                  onClick={() => this.handleToggleChange('notifications_idea_comment')}
                />


              </InputGroupStyled>

              <Button
                text={intl.formatMessage(messages.submit)}
                onClick={this.handleSubmit}
                loading={processing}
              />
            </section>
          </RightColumnStyled>
        </Grid.Row>
      </Grid>
    </div>);
  }
}

ProfileForm.propTypes = {
  className: PropTypes.string,
  onFormSubmit: PropTypes.func.isRequired,
  avatarUpload: PropTypes.func.isRequired,
  onLocaleChangeClick: PropTypes.func.isRequired,
  userData: PropTypes.object,
  avatarUploadError: PropTypes.bool,
  processing: PropTypes.bool,
  storeErrors: PropTypes.object,
  locales: ImmutablePropTypes.list,
  intl: intlShape.isRequired,
  tFunc: PropTypes.func.isRequired,
};

const mapStateToProps = createStructuredSelector({
  locales: makeSelectSetting(['core', 'locales']),
});

export default injectTFunc(injectIntl(connect(mapStateToProps)(styled(ProfileForm)`
  width: 80%;
  margin: auto;
`)));
