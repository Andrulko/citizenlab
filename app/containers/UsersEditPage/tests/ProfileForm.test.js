import React from 'react';
import { FormattedMessage } from 'react-intl';
import { createComponentWithIntl, mountWithIntl } from '../../../utils/intlTest';
import messages from '../messages';
import ProfileForm from '../ProfileForm';
import { mapDispatchToProps } from '../index';
import { updateCurrentUser } from '../actions';
import { localStorageMock } from '../../../utils/testUtils';
import { jestFn } from '../../../utils/testConstants';

describe('<ProfileForm />', () => {
  // mock localStorage
  Object.defineProperty(window, 'localStorage', { value: localStorageMock() });

  it('calls onSubmit handler', () => {
    const handleSubmitMock = jest.fn();
    const user = {};

    const wrapper = mountWithIntl(<ProfileForm
      onFormSubmit={handleSubmitMock}
      avatarUpload={jestFn}
      avatarUploadError={jestFn}
      onLocaleChangeClick={jestFn}
      userData={user}
    />);
    wrapper.find('[type="submit"]').get(0).click();
    expect(handleSubmitMock.mock.calls.length).toEqual(1);
  });

  describe('onFormSubmit', () => {
    it('updateCurrentUser action should be dispatched', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);

      const payload = {
        userId: 'anything',
      };

      result.onProfileFormSubmit(payload);

      expect(dispatch).toHaveBeenCalledWith(updateCurrentUser(payload));
    });
  });

  describe('<LabelInputPair />', () => {
    it('input\'s label should render correctly', () => {
      const id = 'first_name';
      const tree = createComponentWithIntl(
        <label htmlFor={id}>
          <FormattedMessage {...messages[id]} />
        </label>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
