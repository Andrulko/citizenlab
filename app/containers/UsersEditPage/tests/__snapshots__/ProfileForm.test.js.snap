exports[`<ProfileForm /> <LabelInputPair /> input's label should render correctly 1`] = `
<label
  htmlFor="firstName">
  <span>
    First Name
  </span>
</label>
`;
